typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,628 @@
+import { ServiceID } from "../services/service";
+import { ServiceContext, GeneratedFileFilepaths } from "../services/service_context";
+import { StaticFileID, FilesArtifactID, ContainerCreationConfig } from "../services/container_creation_config"; 
+import { ContainerRunConfig } from "../services/container_run_config";
+import { newGetLoadLambdaArgs, newGetLambdaInfoArgs, newGetRegisterStaticFilesArgs, newGetRegisterFilesArtifactsArgs, newGetRegisterServiceArgs, newGetStartServiceArgs, newGetGetServiceInfoArgs, newGetRemoveServiceArgs, newGetPartitionServices, newGetPartitionConnections, newGetRepartitionArgs, newGetWaitForEndpointAvailabilityArgs, newGetExecuteBulkCommandsArgs } from "../constructor_calls";
+import { okAsync, errAsync, ResultAsync, Result } from "neverthrow";
+import * as winston from "winston";
+import * as path from "path";
+import * as fs from 'fs';
+import * as fsPromises from "fs/promises";
+import * as grpc from "grpc";
+
+export type PartitionID = string;
+
+// This will always resolve to the default partition ID (regardless of whether such a partition exists in the network,
+//  or it was repartitioned away)
+const defaultPartitionId: PartitionID = "";
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class NetworkContext {
+	private readonly client: ApiContainerServiceClient;
+
+	// The location on the filesystem where this code is running where the enclave data volume is mounted
+	private readonly enclaveDataVolMountpoint: string;
+
+
+    /*
+    Creates a new NetworkContext object with the given parameters.
+    */
+    constructor(
+            client: ApiContainerServiceClient,
+            enclaveDataVolMountpoint: string) {
+                this.client = client;
+                this.enclaveDataVolMountpoint = enclaveDataVolMountpoint;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async loadLambda(
+            lambdaId: LambdaID,
+            image: string,
+            serializedParams: string): Promise<[LambdaContext, Error]> {
+        const args: LoadLambdaArgs = newGetLoadLambdaArgs(lambdaId, image, serializedParams);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING (REMOVE)
+        // // We proxy calls to Lambda modules via the API container, so actually no need to use the response here
+        // _, err := networkCtx.client.LoadLambda(context.Background(), args)
+        // if err !== nil {
+        //     return nil, stacktrace.Propagate(err, "An error occurred loading new module '%v' with image '%v' and serialized params '%v'", lambdaId, image, serializedParams)
+        // }
+
+        const promiseAsyncLoadLambda: Promise<ResultAsync<any, Error>> = new Promise((resolve, _unusedReject) => { //TODO - Repeating this promise format a lot, make a function for it?
+            this.client.executeLambda(args, (_unusedError: grpc.ServiceError, response: any) => {
```

Comment:
Careful!! This is the `loadLambda` method, but we're calling the `executeLambda` client call!